demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -0,0 +1,1974 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file ota_demo_core_mqtt.c
+ * @brief OTA update example using coreMQTT.
+ */
+
+/* Standard includes. */
+#include <assert.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <errno.h>
+
+#include "aws_demo_config.h"
+
+#include "FreeRTOS.h"
+#include "task.h"
+#include "semphr.h"
+
+#include "iot_network.h"
+
+/* CoreMQTT-Agent APIS for running MQTT in a multithreaded environment. */
+#include "freertos_agent_message.h"
+#include "freertos_command_pool.h"
+
+/* CoreMQTT-Agent include. */
+#include "core_mqtt_agent.h"
+
+/* CoreMQTT-Agent configuration file. */
+#include "core_mqtt_agent_config.h"
+
+/* Includes helpers for managing MQTT subscriptions. */
+#include "subscription_manager.h"
```

Comment:
This doesn't match the filename used in `demos/common`, it's the one in `demos/coreMQTT_Agent`

Reply:
Resolved in https://github.com/aws/amazon-freertos/pull/3188/commits/450fda5aa67e04ef89b710317773948af5a9259c